### PR TITLE
Test running openblack on CI

### PR DESCRIPTION
**Pull request by bwrsandman**
Add scripts which generate mock game data sufficient enough for running openblack without Black & White game data.
Add commandline option to run a certain amount of frames.
Make the Noop renderer run without a window.

### CODE REVIEW

**Comment by bwrsandman** [comment]
Appveyor is running python 2... I don't know how to invoke python 3

File: ci/mock/Data/WeatherSystem/gen_sky.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        'Sky_Ntrl_Dusk.555',
+        'Sky_Ntrl_Night.555',
+    ]
+    texture_width = 2
+    texture_height = 2
+
+    model_magic = b'L3D0'
+    with open(os.path.join(dir_name, "sky.l3d"), "wb") as f:
+        f.write(model_magic)
+        f.write(b'\0' * 4 * 18)  # Header
```
Comment (bwrsandman) [inline ci/mock/Data/WeatherSystem/gen_sky.py]: This can be done with l3d tool

File: ci/mock/Data/gen_all_meshes.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        f.write(b'\0' * (INFO_LHBlockHeader_blockSize - 4))
+
+        f.write(MESHES_LHBlockHeader_blockName)
+        f.write(MESHES_LHBlockHeader_blockSize.to_bytes(4, byteorder='little'))
+        f.write(MESHES_magic)
+        f.write(MESHES_count.to_bytes(4, byteorder='little'))
+        f.write(b'\0' * (MESHES_LHBlockHeader_blockSize - len(MESHES_magic) - 4))
+
+if __name__ == '__main__':
+    main(sys.argv)
```
Comment (bwrsandman) [inline ci/mock/Data/gen_all_meshes.py]: This can be done with g3dtool

File: ci/mock/Data/Landscape/gen_land1.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        f.write(lowresCount.to_bytes(4, byteorder='little'))
+
+        f.write(terrain_type.to_bytes(2, byteorder='little'))
+        f.write(material_data)
+
+        f.write(noise_map_data)
+        f.write(bump_map_data)
+
+if __name__ == '__main__':
+    main(sys.argv)
```
Comment (bwrsandman) [inline ci/mock/Data/Landscape/gen_land1.py]: This can be done with lndtool

**Comment by bwrsandman** [comment]
Will rebase this on top of the components PR so to not use python to generate mock meshes and terrain.

**Comment by bwrsandman** [comment]
It's failing due in part due to https://github.com/microsoft/vcpkg/pull/8642

File: ci/mock/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+set(SKY_OUTPUT ${SKY_BINARY_DIR}/sky.l3d)
+set(SKY_SUBMESH_FILE ${SKY_BINARY_DIR}/sky.obj)
+
+set(TERRAIN_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/Data/Landscape)
+set(TERRAIN_TYPE 0)
+set(TERRAIN_OUTPUT ${TERRAIN_BINARY_DIR}/Land1.lnd)
+set(TERRAIN_NOISE_MAP ${CMAKE_CURRENT_SOURCE_DIR}/Data/Landscape/noise_map.r8.raw)
+set(TERRAIN_BUMP_MAP ${CMAKE_CURRENT_SOURCE_DIR}/Data/Landscape/bump_map.r8.raw)
+list(APPEND TERRAIN_MATERIAL_ARRAY ${CMAKE_CURRENT_SOURCE_DIR}/Data/Landscape/material_data1.rgb5a1.raw)
+list(APPEND TERRAIN_MATERIAL_ARRAY ${CMAKE_CURRENT_SOURCE_DIR}/Data/Landscape/material_data2.rgb5a1.raw)
```
Comment (bwrsandman) [inline ci/mock/CMakeLists.txt]: This should all be from binary directory

File: azure-pipelines/templates/test-null-renderer.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+steps:
+- bash: |
+    cmake --build build --target generate-mock-game-data
+  displayName: 'Generate mock game data'
+- bash: |
+    cp $PWD/bin/${{ parameters.BuildConfiguration }}/fmtd.dll /c/vcpkg/installed/${{ parameters.BuildPlatform }}-windows/bin/SDL2.dll .
```
Comment (bwrsandman) [inline azure-pipelines/templates/test-null-renderer.yml]: I really don't like having to do this.

There must be a better way...

**Comment by handsomematt** [comment]
only real problem is the copying of the dlls, which should be handled by vcpkg but isn't being here - future issue.

**Pull request merged**